### DISCUSSION TITLE
what changed do I need to do to the example cdc_dual_ports on RP2040 to have only one Serial Port instead of two?

### DISCUSSION THREAD

**Discussion by nachiketathakur** [hathach/tinyusb#1415] [Q&A]
I am trying to have CDC VCP on RP2040(Pico Development board). The example cdc_dual_ports enable two ports for serial communication. For my application, I want just one serial port. I have tried setting the macro `CFG_TUD_CDC` to 1, but it is not working. I have also commented out second CDC interface, but that also won't work. What am I missing? Can someone point me to right direction? 
 ```
 // 2nd CDC: Interface number, string index, EP notification address and size, EP data address (out, in) and size.
 // TUD_CDC_DESCRIPTOR(ITF_NUM_CDC_1, 4, EPNUM_CDC_1_NOTIF, 8, EPNUM_CDC_1_OUT, EPNUM_CDC_1_IN, 64),
```

**Comment by hathach** [comment]
could you give link to your fork modified example where other could comment on line of code for suggestion ?

**Reply by nachiketathakur** [reply]
Hello.
The link to the example is: https://github.com/hathach/tinyusb/blob/9c8c5c1c53214b8954bb28c33a496b5b423d5ecc/examples/device/cdc_dual_ports/src/main.c
I just downloaded it and modifying the example in my local project.

**Reply by hathach** [reply]
I mean what have you modified to the descriptors, made it public on github so I could point out what is the issue.

**Reply by nachiketathakur** [reply]
Can you please point me to the section of port which initializes two serial virtual com ports?

**Reply by hathach** [reply]
USB interfaces is dictated by USB descriptors, you should spend time to learn about usb descriptors.

**Reply by nachiketathakur** [reply]
Thank you for the inputs, I was able to open the single port only. 

> USB interfaces is dictated by USB descriptors, you should spend time to learn about usb descriptors.
